src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -253,6 +256,42 @@ private static string GetTestAgentExePath(bool requires32Bit)
             return Path.Combine(engineDir, agentName);
         }
 
+        private static void OnAgentExit(object sender, EventArgs e)
+        {
+            var process = sender as Process;
+            if (process == null)
+                return;
+
+            string errorMsg;
+
+            switch (process.ExitCode)
+            {
+                case AgentExitCodes.OK:
+                    return;
+                case AgentExitCodes.PARENT_PROCESS_TERMINATED:
+                    errorMsg = "Remote test agent believes agency process has exited.";
```

Comment:
😸 Obviously, we don't ever expect to see this message - belt and suspenders though.

Reply:
I included all the existing exit points into this functionality. This one seemed odd, but I knew if I didn't write a custom message, then it would be the first to come up! 😆